x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/index.tsx

```diff
@@ -6,38 +6,50 @@
  */
 
 import { EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiText, EuiToolTip } from '@elastic/eui';
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useCallback, useMemo } from 'react';
 
 import { css } from '@emotion/react';
+import { isEmpty } from 'lodash/fp';
 import { useAssistantContext } from '../../../assistant_context';
-import { Conversation, Prompt } from '../../../..';
+import { Conversation } from '../../../..';
 import * as i18n from './translations';
 import { SelectSystemPrompt } from './select_system_prompt';
 
 interface Props {
   conversation: Conversation | undefined;
+  editingSystemPromptId: string | undefined;
+  isSettingsModalVisible: boolean;
+  onSystemPromptSelectionChange: (systemPromptId: string | undefined) => void;
+  setIsSettingsModalVisible: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
-const SystemPromptComponent: React.FC<Props> = ({ conversation }) => {
+const SystemPromptComponent: React.FC<Props> = ({
+  conversation,
+  editingSystemPromptId,
+  isSettingsModalVisible,
+  onSystemPromptSelectionChange,
+  setIsSettingsModalVisible,
+}) => {
   const { allSystemPrompts } = useAssistantContext();
 
-  const [selectedPrompt, setSelectedPrompt] = useState<Prompt | undefined>(
-    allSystemPrompts?.find((p) => p.id === conversation?.apiConfig.defaultSystemPromptId) ??
-      allSystemPrompts?.[0]
-  );
-
-  useEffect(() => {
-    setSelectedPrompt(
-      allSystemPrompts?.find((p) => p.id === conversation?.apiConfig.defaultSystemPromptId) ??
-        allSystemPrompts?.[0]
-    );
-  }, [allSystemPrompts, conversation]);
+  const selectedPrompt = useMemo(() => {
+    if (editingSystemPromptId !== undefined) {
+      return (
+        allSystemPrompts?.find((p) => p.id === editingSystemPromptId) ??
```

Comment:
This prefers the system prompt selected by the user in the editor, falls back to the conversation's system prompt (if it can be found), or returns `undefined` if none of the above yield a valid system prompt.